File: app/src/main/java/com/example/breezil/favnews/ui/adapter/PagerAdapter.java
package com.example.breezil.favnews.ui.adapter;

import android.content.Context;
import android.content.res.Resources;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;
import android.util.SparseArray;
import android.view.ViewGroup;

import com.example.breezil.favnews.R;
import com.example.breezil.favnews.ui.BookMarksFragment;
import com.example.breezil.favnews.ui.HeadlinesFragment;
import com.example.breezil.favnews.ui.PreferredFragment;


public class PagerAdapter extends FragmentStatePagerAdapter {
    Context context;
    SparseArray<Fragment> fragmentList = new SparseArray<>();

    private static final int preferred_Position = 0;
    private static final int headline_Position = 1;
    private static final int bookmark_Position = 2;


    public PagerAdapter(FragmentManager fm, Context context) {
        super(fm);
        this.context = context;
    }

    @Override
    public Fragment getItem(int position) {
        if(position == preferred_Position){
            return new PreferredFragment();
        }else if(position == headline_Position){
            return new HeadlinesFragment();
        }else {
            return new BookMarksFragment();
        }
    }
    @Override
    public int getCount() {
        return 3;
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        Resources resources = context.getResources();
        switch (position){
            case preferred_Position:
                return resources.getString(R.string.preferred_fragment);

            case headline_Position:
                return resources.getString(R.string.head_line_fragment);
            case bookmark_Position:
                return resources.getString(R.string.book_mark_fragment);
            default:
                return "none";

        }
    }
    @NonNull
    @Override
    public Object instantiateItem(ViewGroup container, int position) {
        Fragment fragment = (Fragment) super.instantiateItem(container,position);
        fragmentList.put(position, fragment);
        return fragment;
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object object) {
        fragmentList.remove(position);
        super.destroyItem(container,position,object);
    }
}

File: app/src/main/java/com/fave/breezil/fave/ui/adapter/PagerAdapter.kt
package com.fave.breezil.fave.ui.adapter

import android.content.Context
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentStatePagerAdapter
import android.util.SparseArray
import android.view.ViewGroup

import com.fave.breezil.fave.R
import com.fave.breezil.fave.ui.main.fragment.BookMarksFragment
import com.fave.breezil.fave.ui.main.fragment.HeadlinesFragment
import com.fave.breezil.fave.ui.main.fragment.PreferredFragment


class PagerAdapter(fm: FragmentManager, internal var context: Context) : FragmentStatePagerAdapter(fm) {
    private var fragmentList = SparseArray<Fragment>()

    override fun getItem(position: Int): Fragment {
        return when (position) {
            preferred_Position -> PreferredFragment()
            headline_Position -> HeadlinesFragment()
            else -> BookMarksFragment()
        }
    }

    override fun getCount(): Int {
        return 3
    }

    override fun getPageTitle(position: Int): CharSequence? {
        val resources = context.resources
        return when (position) {
            preferred_Position -> resources.getString(R.string.preferred_fragment)

            headline_Position -> resources.getString(R.string.head_line_fragment)
            bookmark_Position -> resources.getString(R.string.book_mark_fragment)
            else -> "none"
        }
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val fragment = super.instantiateItem(container, position) as Fragment
        fragmentList.put(position, fragment)
        return fragment
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        fragmentList.remove(position)
        super.destroyItem(container, position, `object`)
    }

    companion object {

        private const val preferred_Position = 0
        private const val headline_Position = 1
        private const val bookmark_Position = 2
    }
}
